src/main/java/edu/harvard/iq/dataverse/ingest/IngestUtil.java

```diff
@@ -257,7 +257,7 @@ public static String generateNewFileName(final String fileName) {
         // #6942 added proxy for existing files to a boolean set when dataset version copy is done
         for (Iterator<FileMetadata> fmIt = version.getFileMetadatas().iterator(); fmIt.hasNext();) {
             FileMetadata fm = fmIt.next();
-            if((fm.isInPriorVersion() || fm.getId() != null) && (replacedFmd==null) || (!fm.getDataFile().equals(replacedFmd.getDataFile()))) {
+            if((fm.isInPriorVersion() || fm.getId() != null) && (replacedFmd==null || !fm.getDataFile().equals(replacedFmd.getDataFile()))) {
```

Comment:
Is this related to this PR?

Reply:
Implemented this change to solve the below exception. It got resolved,  but have to confirm with the team.


[2021-05-28T11:47:00.302-0400] [Payara 5.2021.1] [WARNING] [AS-EJB-00056] [javax.enterprise.ejb.container] [tid: _ThreadID=95 _ThreadName=http-thread-pool::http-listener-1(2)] [timeMillis: 1622216820302] [levelValue: 900] [[
  A system exception occurred during an invocation on EJB IngestServiceBean, method: public java.util.List edu.harvard.iq.dataverse.ingest.IngestServiceBean.saveAndAddFilesToDataset(edu.harvard.iq.dataverse.DatasetVersion,java.util.List,edu.harvard.iq.dataverse.DataFile)]]

javax.ejb.EJBException
	at com.sun.ejb.containers.EJBContainerTransactionManager.processSystemException(EJBContainerTransactionManager.java:723)
	at com.sun.ejb.containers.EJBContainerTransactionManager.completeNewTx(EJBContainerTransactionManager.java:652)
	at com.sun.ejb.containers.EJBContainerTransactionManager.postInvokeTx(EJBContainerTransactionManager.java:482)
	at com.sun.ejb.containers.BaseContainer.postInvokeTx(BaseContainer.java:4592)
	at com.sun.ejb.containers.BaseContainer.postInvoke(BaseContainer.java:2125)
	at com.sun.ejb.containers.BaseContainer.postInvoke(BaseContainer.java:2095)
	at com.sun.ejb.containers.EJBLocalObjectInvocationHandler.invoke(EJBLocalObjectInvocationHandler.java:220)
	at com.sun.ejb.containers.EJBLocalObjectInvocationHandlerDelegate.invoke(EJBLocalObjectInvocationHandlerDelegate.java:90)
	at com.sun.proxy.$Proxy352.saveAndAddFilesToDataset(Unknown Source)
	**at edu.harvard.iq.dataverse.ingest.__EJB31_Generated__IngestServiceBean__Intf____Bean__.saveAndAddFilesToDataset(Unknown Source)**
	at edu.harvard.iq.dataverse.datasetutility.AddReplaceFileHelper.step_060_addFilesViaIngestService(AddReplaceFileHelper.java:1563)
	at edu.harvard.iq.dataverse.datasetutility.AddReplaceFileHelper.runAddReplacePhase2(AddReplaceFileHelper.java:736)
	at edu.harvard.iq.dataverse.datasetutility.AddReplaceFileHelper.runAddReplaceFile(AddReplaceFileHelper.java:489)
	at edu.harvard.iq.dataverse.datasetutility.AddReplaceFileHelper.runAddFileByDataset(AddReplaceFileHelper.java:349)
	at edu.harvard.iq.dataverse.datasetutility.AddReplaceFileHelper.addFiles(AddReplaceFileHelper.java:2036)
	at edu.harvard.iq.dataverse.api.Datasets.addFilesToDataset(Datasets.java:2507)
	……………..
**Caused by: java.lang.NullPointerException
	at edu.harvard.iq.dataverse.ingest.IngestUtil.existingPathNamesAsSet(IngestUtil.java:260)**
	at edu.harvard.iq.dataverse.ingest.IngestUtil.checkForDuplicateFileNamesFinal(IngestUtil.java:66)
	at edu.harvard.iq.dataverse.ingest.IngestServiceBean.saveAndAddFilesToDataset(IngestServiceBean.java:167)